app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -27,7 +27,10 @@ def global_registration_vars
                    organization: (organization || User.current.default_organization || User.current.my_organizations.first),
                    location: (location || User.current.default_location || User.current.my_locations.first),
                    hostgroup: host_group,
-                   operatingsystem: operatingsystem })
+                   operatingsystem: operatingsystem,
+                   url_host: registration_url.host,
+                   registration_url: registration_url,
+                    })
```

Comment:
Not strictly needed and certainly not in this PR, but I feel like this hash is getting long. Perhaps it would be good to spit it up:

```ruby
context = {
  user: User.current,
  # .. other ones
}

params.permit(permitted).to_h.symbolize_keys.merge(context)
```